### DISCUSSION TITLE
Integration of OpenMP/MPI for MET_COMPILE_ALL.sh

### DISCUSSION THREAD

**Discussion by ghost** [dtcenter/METplus#1678] [Installation] [answered]
Good afternoon METplus team,

I was wondering if there is a way to add parallelization installation option for the MET_Compile_all.sh script.  As it stands the dependencies are installed in serial as well as MET which can take several minutes to over half an hour or more due to system resources. 

By utilizing multiple processors this installation could be cut down by 25-50% ideally.  

Would this be something to consider if funding is available for it?

Also I have noticed some of the dependent libraries are legacy libraries; (NETCDF mainly) will these be updated in the future?

**Comment by ghost** [comment]
So by looking at the compile_met_all script.  mpicc can be called as an argument in front of the script ($FC $CC $CXX etc.).  However the compile script is not parallel enabled with (-j (# of cores) ).  So mpich would have to be installed ahead of running compile_met_all.sh

**Comment by jprestop** [comment, marked as answer]
Hi @whatheway.  Thank you for this suggestion. This is definitely something to consider if there is funding available for it.  I have created the following issue in the MET repository for this potential improvement: [Consider adding parallelization to the compile_MET_all.sh script #2200](https://github.com/dtcenter/MET/issues/2200).

If you have additional notes to add, please add them to the issue in the link above. 

Regarding the following question:

> Also I have noticed some of the dependent libraries are legacy libraries; (NETCDF mainly) will these be updated in the future?

MET isn't dependent on any particular version of NetCDF.  Users are able to install MET with whatever version of NetCDF they'd like. We do include a specific version in the tar_files.tgz package, which I will plan on updating for the next release.  Thank you for the suggestion.  

I'd like to mark this discussion as being answered and "lock it" to prevent future posts. That's how we encourage users to create new discussions for new questions. But I wanted to give you an opportunity to comment on it before doing so.  For further discussion on the suggested improvements, we can continue to comment on the issue above.
